slicedimage/_tile.py

```diff
@@ -2,28 +2,41 @@
 
 import warnings
 
-from ._typeformatting import format_tile_coordinates, format_tile_indices
+from ._typeformatting import format_enum_keyed_dicts, format_tile_coordinates
 
 
 class Tile(object):
     def __init__(self, coordinates, indices, tile_shape=None, sha256=None, extras=None):
         self.coordinates = format_tile_coordinates(coordinates)
-        self.indices = format_tile_indices(indices)
-        self._tile_shape = tuple(tile_shape) if tile_shape is not None else None
+        self.indices = format_enum_keyed_dicts(indices)
+        self._tile_shape = format_enum_keyed_dicts(tile_shape) if tile_shape is not None else None
         self.sha256 = sha256
         self.extras = {} if extras is None else extras
 
         self._numpy_array = None
         self._numpy_array_future = None
 
+    @staticmethod
+    def format_tuple_shape_to_dict_shape(tuple_shape):
+        if tuple_shape is None:
+            return None
+        return {'y': tuple_shape[-2], 'x': tuple_shape[-1]}
```

Comment:
why do you use the negative indexing here?

Reply:
rationale: even though we've never done it, it is theoretically possible to use N-dimensional chunking, where N>2.  This future-proofs the code, because it is assumed that the higher dimension would be the most significant in iteration order.